packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -605,6 +609,55 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
     public async uploadSummaryWithContext(summary: api.ISummaryTree, context: ISummaryContext): Promise<string> {
         this.checkSnapshotUrl();
 
+        let retry = 0;
+        for (;;) {
```

Comment:
Could we have this only run when the client is uploading the whole summary tree. Ideally we would not have to do this for the existing summary system (2 stage)